invenio_users_resources/services/users/service.py

```diff
@@ -57,14 +75,42 @@ def create(self, identity, data, raise_errors=True, uow=None):
             errors=errors,
             uow=uow,
         )
-
-        # persist user to DB (indexing is done in the session hooks, see ext)
-        uow.register(RecordCommitOp(user))
-
+        # get email token and reset info
+        account_user = current_datastore.get_user(user.id)
+        token, reset_link = default_reset_password_link_func(account_user)
+        # trigger celery task to send email.
+        uow.register(
+            TaskOp(
+                execute_reset_password_email,
+                user_id=user.id,
+                token=token,
+                reset_link=reset_link,
+            )
+        )
         return self.result_item(
             self, identity, user, links_tpl=self.links_item_tpl, errors=errors
         )
 
+    def _generate_password(self, length=12):
+        """Generate password of a specific length."""
+        alphabet = string.ascii_letters + string.digits
+        return "".join(secrets.choice(alphabet) for _ in range(length))
+
+    def _create_user_as_admin(
+        self,
+        user_info: dict,
+    ):
+        """Create a new active and verified user with auto-generated password."""
+        # Generate password and add to user_info dict
+        user_info["password"] = hash_password(self._generate_password())
+
+        # Create the user with the specified data
+        user = self.user_cls.create(user_info)
+        # Activate and verify user
+        user.activate()
+        user.verify()
```

Comment:
We thought/discussed this internally and concluded that, because an email to change the password has already been sent, the email would be verified and correct once the new user changed their password. Does it make sense?